### PR TITLE
[Demo] React PP Demo

### PR DESCRIPTION
**Pull request by rohanzemoso**
# React Pair Programming Demo
## What
- React Pair Programming Demo on Minnet Application's component
## How
- Demonstration using react TS, MUI and atomic design pattern.
## Why
- To Demonstrate React TS Frontend Skills during React Pair Programming

## Screenshots:
image

image

### CODE REVIEW

File: Minnet App/src/components/molecules/TypographyIcon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}));
+
+const StyledIconTypographyBox = styled(Box)(() => ({
+  display: "flex",
+  gap: "1rem",
+  alignItems: "center",
+}));
+
+export const TypographyIcon: React.FC<TypographyIcon_Props> = ({
+  ...props
```
Comment (mahitha-29) [inline Minnet App/src/components/molecules/TypographyIcon/index.tsx:46]: The current name doesn't accurately reflect the purpose of the component. Consider using a more descriptive and relevant name, such as TradeRow or something similar

File: Minnet App/src/components/molecules/TypographyIcon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}) => (
+  <StyledBox>
+    <StyledIconTypographyBox>
+      <IconAtom src={props.src} />
+      <TypographyMolecule name={props.name} subtitle={props.subtitle} />
```
Comment (mahitha-29) [inline Minnet App/src/components/molecules/TypographyIcon/index.tsx:51]: The Icon and Typography above can be combined into a separate reusable molecule component.

File: Minnet App/src/components/molecules/TypograpyMolecule/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Box } from "@mui/material";
+import React from "react";
+import TypographyAtom from "../../atoms/Typography";
+import { theme } from "../../../theme/theme";
+
+interface TypographyMolecule_Props {
+  name: string;
+  subtitle: string;
+}
+
```
Comment (mahitha-29) [inline Minnet App/src/components/molecules/TypograpyMolecule/index.tsx:10]: This molecule is not required

File: Minnet App/src/components/atoms/StarIcon/index.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { IconButton } from "@mui/material";
+import StarIcon from "@mui/icons-material/Star";
+import StarBorderIcon from "@mui/icons-material/StarBorder";
+
+interface StarIcon_Props {
+  isWatched: boolean;
+  onClick: () => void;
+}
+
```
Comment (mahitha-29) [inline Minnet App/src/components/atoms/StarIcon/index.tsx:10]: Try to use the same naming convention for folder and the component

**Review by mahitha-29 — CHANGES_REQUESTED** [review]

File: Minnet App/src/components/molecules/TypographyIcon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  subtitle: string;
+  price: string;
+  Change: string;
+  MarketCap: string;
+  isWatched: boolean;
+  onWatchToggle: () => void;
+}
+
+const StyledBox = styled(Box)(() => ({
+  width: "1100px",
```
[the remark this answers is not in the document]
Comment (rohanzemoso) [reply Minnet App/src/components/molecules/TypographyIcon/index.tsx:22]: changed pixel to theme.spacing, resolved.

File: Minnet App/src/components/molecules/TypographyIcon/index.tsx (hunk 2 above)
Comment (rohanzemoso) [reply Minnet App/src/components/molecules/TypographyIcon/index.tsx:51]: sure, separated the said components for reusability

File: Minnet App/src/components/atoms/StarIcon/index.tsx (hunk 4 above)
Comment (rohanzemoso) [reply Minnet App/src/components/atoms/StarIcon/index.tsx:10]: apologies, used proper naming convention now. resolved.

File: Minnet App/src/components/organisms/AssetRows/index.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  const displayRows =
+    tabValue === 1 ? rows.filter((item) => watchedItems.has(item.name)) : rows;
+
+  return (
+    <StyledBox>
+      <TabsMolecule value={tabValue} onChange={handleTabChange} />
+      <Header />
+      {displayRows.map((item, index) => (
+        <TradeRow
+          key={index}
```
Comment (mahitha-29) [inline Minnet App/src/components/organisms/AssetRows/index.tsx:43]: Avoid using index , instead add id in the data and use it here for key

File: Minnet App/src/utils/constants/index.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+const path = "assets/icons";
+
+
+
+export const rows = [
+    {
```
Comment (mahitha-29) [inline Minnet App/src/utils/constants/index.ts:6]: Try to give descriptive name

**Review by mahitha-29 — CHANGES_REQUESTED** [review]

File: Minnet App/src/components/organisms/AssetRows/index.tsx (hunk 6 above)
Comment (rohanzemoso) [reply Minnet App/src/components/organisms/AssetRows/index.tsx:43]: added id in data and using as key, resolved.

File: Minnet App/src/utils/constants/index.ts (hunk 7 above)
Comment (rohanzemoso) [reply Minnet App/src/utils/constants/index.ts:6]: gave proper descriptive name to the array of objects, resolved.

**Review by mahitha-29 — APPROVED** [review]

**Pull request merged**